### PR TITLE
gcs: autotune: sharing enhancements and fixes

### LINKED ISSUE #363: Autotune: would be really nice to gather more data
@dustin @tracernz @droidicus 

When we're studying Autotune for developer improvement, more data would be really useful.

At minimum,

1. All the contents of the systemident object.  Raw per-axis betas (yaw is missing now), number of spilled points.
2. Actuator curve setting

### PR DESCRIPTION
**Pull request by mlyle**
Fixes #363 : gathers various uav object contents

Also fixes the noise sensitivity bug where it's always been shared as 0.

[Review on Reviewable](https://reviewable.io/reviews/d-ronin/dronin/382)

### CODE REVIEW

File: ground/gcs/src/plugins/config/configautotunewidget.cpp [hunk 1]
```diff
@@ -162,6 +163,11 @@ void ConfigAutotuneWidget::onShareToDatabase()
 
     QJsonDocument json = getResultsJson();
 
+#if 0
+    QString jsonString = json.toJson(QJsonDocument::Indented);
+    qDebug() << jsonString;
+#endif
+
```
Comment (PTDreamer) [inline ground/gcs/src/plugins/config/configautotunewidget.cpp]: Debug code should be deleted

File: ground/gcs/src/plugins/config/configautotunewidget.cpp [hunk 2]
```diff
@@ -544,15 +581,22 @@ QJsonDocument ConfigAutotuneWidget::getResultsJson()
...
+    identification["yaw"] = yaw_ident;
+
     identification["tau"] = m_autotune->rollTau->text().toDouble();
     json["identification"] = identification;
 
-    QJsonObject tuning, parameters, computed;
+    QJsonObject tuning, parameters, computed, misc;
     parameters["damping"] = m_autotune->lblDamp->text().toDouble();
-    parameters["noiseSensitivity"] = m_autotune->lblNoise->text().toDouble();
+    parameters["noiseSensitivity"] = m_autotune->lblNoise->text().split(" ").at(0).toDouble();
```
Comment (PTDreamer) [inline ground/gcs/src/plugins/config/configautotunewidget.cpp]: I have been bitten by the "at" function before. If for some reason the "split" returns an empty list the "at" will segfault. Although cumbersome the size should be checked before "at" is used.

**Comment by PTDreamer** [comment]
Reviewed 5 of 5 files at r1.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[ground/gcs/src/plugins/config/autotune.ui, line 82 \[r1\]](https://reviewable.io:443/reviews/d-ronin/dronin/382#-K784pERO3WE17Q_0i73:-K784pERO3WE17Q_0i74:2108808777) ([raw file](https://github.com/d-ronin/dronin/blob/65bb1e20c85bfbe0d659b382e7884a19f71f5088/ground/gcs/src/plugins/config/autotune.ui#L82)):*
Was the font change on purpose or just because the file was edited on a different OS?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/d-ronin/dronin/382)*

**Comment by PTDreamer** [comment]
Code looks good. Haven't tested it though

---

Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/d-ronin/dronin/382#-:-K785QS4PRBPMlmlXFSc:-98532489)*

**Comment by dustin** [comment]
:gtm: with @PTDreamer's comments addressed

---

Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/d-ronin/dronin/382#-:-K787RzyVaZEjZejBAsp:819118086)*

**Comment by dustin** [comment]
Related, the auto expansion thing somehow ate the :lgtm:

---

Review status: all files reviewed at latest revision, 3 unresolved discussions.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/d-ronin/dronin/382#-:-K787ZmcC-vgEqPJJC4x:-277076184)*

**Comment by droidicus** [comment]
Tested with Win10 and Sparky2.
Before:
[clipboard_next.txt](https://github.com/d-ronin/dRonin/files/76735/clipboard_next.txt)
[json_next.txt](https://github.com/d-ronin/dRonin/files/76734/json_next.txt)

After:
[clipboard_next382.txt](https://github.com/d-ronin/dRonin/files/76736/clipboard_next382.txt)
[json_next382.txt](https://github.com/d-ronin/dRonin/files/76737/json_next382.txt)

The Clipboard copy is unchanged (other than my note in the "observations").

**Comment by mlyle** [comment]
Review status: 4 of 5 files reviewed at latest revision, all discussions resolved.

---

*[ground/gcs/src/plugins/config/autotune.ui, line 82 \[r1\]](https://reviewable.io:443/reviews/d-ronin/dronin/382#-K784pERO3WE17Q_0i73:-K78DhXTJH2WhQ_AowYw:2069000985) ([raw file](https://github.com/d-ronin/dronin/blob/65bb1e20c85bfbe0d659b382e7884a19f71f5088/ground/gcs/src/plugins/config/autotune.ui#L82)):*
Yah, QT creator just likes changing this stuff.   Is there a graceful way to deal with this?  It seemed to make no difference in actual UI appeararance.

---

*[ground/gcs/src/plugins/config/configautotunewidget.cpp, line 595 \[r1\]](https://reviewable.io:443/reviews/d-ronin/dronin/382#-K77yM7-yzaB7Y2_3pY6-r1-595:-K78El0O5IQaa1blgIDJ:-499146009) ([raw file](https://github.com/d-ronin/dronin/blob/65bb1e20c85bfbe0d659b382e7884a19f71f5088/ground/gcs/src/plugins/config/configautotunewidget.cpp#L595)):*
Done.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/d-ronin/dronin/382)*

**Comment by PTDreamer** [comment]
Reviewed 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[ground/gcs/src/plugins/config/autotune.ui, line 82 \[r1\]](https://reviewable.io:443/reviews/d-ronin/dronin/382#-K784pERO3WE17Q_0i73:-K78J7nF5czquThIeuZG:-340352862) ([raw file](https://github.com/d-ronin/dronin/blob/65bb1e20c85bfbe0d659b382e7884a19f71f5088/ground/gcs/src/plugins/config/autotune.ui#L82)):*
No idea sorry. About it looking the same I think it falls back to system default if the font doesn't exist on the target system.

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/d-ronin/dronin/382)*

**Comment by PTDreamer** [comment]
:lgtm: 

---

Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/d-ronin/dronin/382#-:-K78KWPR-3fyQBL1EU3S:980077423)*

**Comment by tracernz** [comment]
Now the autotune data format is tied to UAVO version. Not sure if that's good..

**Comment by mlyle** [comment]
@tracernz -- Only the "new stuff" is tied to UAVO version.  All the existing fields are there "normalized" as before.

Basically, I want to be able to survey random parameters over the life of a release.  Like look at all tunes using actuator curve fitting or tell whether people are using oneshot.  I don't want to a priori think of what fields to include and keep it maintained.

I'm fine that this may drift and I may not be able to do meaningful queries arbitrarily far back in the past.  (This would be true if we explicitly maintained a separate format of attributes, too).

**Comment by dustin** [comment]
We've got sample data.  I'm not sure if we want to normalize it any in storage, but the raw stuff is there, so I think it's pretty good.

---

Review status: all files reviewed at latest revision, 1 unresolved discussion, all commit checks successful.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/d-ronin/dronin/382#-:-K79Fqjk8VE8Gop3WUu5:-1002079725)*

**Comment by tracernz** [comment]
Should send the uavo hash for the release then I think.

**Comment by mlyle** [comment]
Hmm. I send the IDs, but you're right, that would have been good.

**Comment by tracernz** [comment]
Should penalty increment the data version too.

**Comment by tracernz** [comment]
Probably*, auto correct :-/

**Pull request merged**